Tests/Functional/Drivers/Queue/PhpAmqpLibDriverTest.php

```diff
@@ -0,0 +1,259 @@
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->ack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the consume process with a callback in return that receives the messages and nack this after
+     *
+     * @dataProvider getMessages
+     */
+    public function testConsumeWithCallbackNackingMessage(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $amqpMessage = null;
+
+        $callback = function($message) use (&$amqpMessage) {
+            $amqpMessage =  $message;
+            $queueMessage = new QueueMessage();
+            $queueMessage->setMessageId($amqpMessage->getDeliveryTag());
+            $this->driver->nack($queueMessage);
+        };
+
+        $this->driver->consume($this->consumer->getName(), $this->queueName, $callback);
+        $this->driver->declareChannel()->is_consuming();
+        $this->driver->wait();
+
+        $this->assertInstanceOf(AMQPMessage::class, $amqpMessage);
+        $this->assertEquals($amqpMessage->delivery_info['routing_key'], $this->queueName);
+        $this->assertEquals($amqpMessage->delivery_info['consumer_tag'], $this->consumer->getName());
+        $this->assertNotNull($amqpMessage->getBody());
+    }
+
+    /**
+     * Tests the function to destroy the connection and clean the variables related
+     *
+     * @dataProvider getMessages
+     * @group destroy
+     */
+    public function testDestroy(MessageInterface $msg)
+    {
+        $this->prepareToConsume($msg);
+        $this->driver->declareChannel();
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->consume($this->consumer->getName(), $this->queueName);
+        $this->driver->destroy($this->consumer->getName());
+        $this->assertFalse($this->driver->isConnected());
+    }
+
+     /**
+     * Tests that tries to declare a channel without having a connection
+     */
+    public function testDeclareChannelWithoutConnnection()
+    {
+        $this->expectException(AMQPConnectionClosedException::class);
+        $this->expectExceptionMessage('Broken pipe or closed connection');
+        $this->assertTrue($this->driver->isConnected());
+        $this->driver->declareChannel();
+        $this->driver->destroy($this->createConsumer()->getName());
+        $this->assertFalse($this->driver->isConnected());
+        $this->driver->declareChannel();
```

Comment:
This test could also be simplified to something like this
```
$this->expectException(AMQPConnectionClosedException::class);
$this->expectExceptionMessage('Broken pipe or closed connection');

$driver = $this->createDriver();
$this->assertFalse($driver->isConnected());
$this->driver->declareChannel();
```

You don't need to continue a positive case to close and then start the negative one. We can have tests just with the negative case

Reply:
It does not work like this. The tests starts with a "driver->connect", so the test starts with a valid connection and I have to disconnect before try to reconnect again. Even if the function $this->createDriver() is used. So, I just enjoy the already state of the class to confirm that before we had a connection and after it was destroyed.

Reply:
Test changed to not depend on setUp status, following the same way as `testConnectWithoutData`

Reply:
Ok

Reply:
Test changed to not depend on setUp status, following the same way as testConnectWithoutData